qos/evm/observe.go

```diff
@@ -0,0 +1,69 @@
+// evm package provides the support required for interacting
+// with an EVM blockchain through the gateway.
+package evm
+
+import (
+	"context"
+)
+
+// observationSet provides all the functionality required
+// by the message package's ObservationSet to handle the sharing
+// of QoS data between PATH instances, and the updating of local
+// PATH instance's QoS data on EVM endpoints.
+var _ message.ObservationSet = observationSet{}
+
+// observation captures the result of processing an endpoint's
+// response to a service request.
+// It provides details needed to establish the validity of
+// an endpoint.
+type observation struct {
+	// TODO_IMPROVE: use a custom type here.
```

Comment:
Thoughts on a protobuf?

Reply:
Agreed it may be a better choice, I have added a `TODO_UPNEXT`